File: go/difference-of-squares/difference_of_squares.go
// Package diffsquares implements utility functions to calculate the
// difference between the square of the sum and the sum of the
// squares of N natural numbers.
package diffsquares

// SquareOfSum calculates the square of the sum for the first n natural numbers.
// See Faulhaber's formula.
func SquareOfSum(n int) int {
	return ((n * (n + 1)) / 2) * ((n * (n + 1)) / 2)
}

// SumOfSquares calculates the sum of the square of the first n natural numbers.
// See Faulhaber's formula.
func SumOfSquares(n int) int {
	return (n * (n + 1) * (2*n + 1)) / 6
}

// Difference calculates the difference of SquareOfSum and SumOfSquares output.
func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}

File: python/difference-of-squares/difference_of_squares.py
def square_of_sum(number: int) -> int:
    """Return sum of first N natural numbers squared.

    This algorithm is based on Faulhaber's formula.
    """
    return ((number * (number + 1)) / 2) ** 2


def sum_of_squares(number: int) -> int:
    """Return sum of the squares of the first N natural numbers.

    This algorithm is based on Faulhaber's formula.
    """
    return (number * (number + 1) * (2 * number + 1)) / 6


def difference_of_squares(number: int) -> int:
    """Return the difference bewteen square of the sum and the sum of the squares."""
    return square_of_sum(number) - sum_of_squares(number)
